src/sentry/snuba/entity_subscription.py

```diff
@@ -0,0 +1,274 @@
+        if not extra_fields or "org_id" not in extra_fields:
+            raise InvalidQuerySubscription(
+                "org_id is a required param when "
+                "building snuba filter for a metrics subscription"
+            )
+        self.org_id = extra_fields["org_id"]
+
+    def build_snuba_filter(
+        self,
+        query: str,
+        environment: Optional[Environment],
+        params: Optional[Mapping[str, Any]] = None,
+    ) -> Filter:
+        resolve_func = resolve_column(Dataset.Sessions)
+        aggregations = [self.aggregate]
+        # This aggregation is added to return the total number of sessions in crash
+        # rate alerts that is used to identify if we are below a general minimum alert threshold
+        count_col = re.search(r"(sessions|users)", self.aggregate)
+        if not count_col:
+            raise UnsupportedQuerySubscription(
+                "Only crash free percentage queries are supported for subscriptions"
+                "over the sessions dataset"
+            )
+        count_col_matched = count_col.group()
+
+        aggregations += [f"identity({count_col_matched}) AS {CRASH_RATE_ALERT_SESSION_COUNT_ALIAS}"]
+        functions_acl = ["identity"]
+        snuba_filter = get_filter(query, params=params)
+        snuba_filter.update_with(
+            resolve_field_list(
+                aggregations, snuba_filter, auto_fields=False, functions_acl=functions_acl
+            )
+        )
+        snuba_filter = resolve_snuba_aliases(snuba_filter, resolve_func)[0]
+        if environment:
+            snuba_filter.conditions.append(["environment", "=", environment.name])
+        return snuba_filter
+
+    def get_entity_extra_params(self) -> Mapping[str, Any]:
+        return {"organization": self.org_id}
+
+
+class MetricsCountersEntitySubscription(BaseEntitySubscription):
+    dataset = Dataset.Metrics.value
+    entity_key = EntityKey.MetricsCounters.value
+
+    def __init__(self, aggregate: str, extra_fields: Optional[Mapping[str, Any]] = None):
+        super().__init__(aggregate, extra_fields)
+        self.aggregate = aggregate
+        if not extra_fields or "org_id" not in extra_fields:
+            raise InvalidQuerySubscription(
+                "org_id is a required param when "
+                "building snuba filter for a metrics subscription"
+            )
+        self.org_id = extra_fields["org_id"]
+
+    def get_query_groupby(self) -> List[str]:
+        session_status = tag_key(self.org_id, "session.status")
+        return ["project_id", session_status]
+
+    def build_snuba_filter(
+        self,
+        query: str,
+        environment: Optional[Environment],
+        params: Optional[Mapping[str, Any]] = None,
+    ) -> Filter:
+        snuba_filter = get_filter(query, params=params)
+        conditions = copy(snuba_filter.conditions)
+        snuba_filter.update_with(
+            {
+                "aggregations": [["sum(value)", None, "value"]],
```

Comment:
This class is called `MetricsCountersEntitySubscription`, but it queries a very specific metric (`session`). Should we rename it to `SessionCrashRateSubscription`? Or are you planning to do the dispatch of future use cases inside this class?

Reply:
>to do the dispatch of future use cases inside this class?

Yes that is exactly it. I am mirroring what I did in snuba for the entity subscription classes